# Data Science Project

In [3]:
library(tidyverse)
library(repr)
library(tidymodels)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

In [4]:
url <- "https://drive.google.com/uc?export=download&id=1kBbW3nLa_kBrNruxi2itBmCwvUT29w8b" 
players_df <- read_csv(url)
head(players_df)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

## Introduction:
### Data Description:

- experience (categorical): the players gaming experience level (Pro, Veteran, Amateur, beginner, or regular)
- subscribe (categorical): whether or not the player is subscribed to the newsletter. (True = subscriber, False = nonsubscriber)
- hasedEmail (categorical): an encrypted version of an email address that protects user privacy
- played_hours (numerical): the total hours played by the player
- name (categorical): players name
- gender (categorical): the players gender identity
- Age (numerical): the players age (in years)

  The “Players” dataset consists of 196 observations (rows) and 7 variables (columns), but there are potential data issues that may impact the accuracy and reliability of the analysis. The variable “Age” has two missing values that should be removed due to incomplete demographic data. Additionally, the “played_hours” variable has a wide distribution of values, consisting of outliers. Most values range from 0 to 50 hours, but some individuals have up to 200+ hours. Larger values could skew the data and lead to an inaccurate analysis. The mean value of each numerical column was found. For played_hours, the mean value was 5.845918 hours and the average age amongst the individuals was 20.52062 years old.



## Methods:
### Data Description:

In [6]:
#finding the average values
players_average_values <- players_df |>
select(played_hours, Age) |>
summarize(across(played_hours:Age, mean, na.rm = TRUE))
players_average_values

#data wrangling
players_data_new <- players_df|>
rename(hashed_email = hashedEmail, age = Age) |>
filter(played_hours < 150.0, age != "NA") |>
arrange(desc(played_hours))
head(players_data_new)

played_hours Age     
1 5.845918     20.52062

experience subscribe
1 Amateur    TRUE     
2 Amateur    TRUE     
3 Amateur    TRUE     
4 Amateur    TRUE     
5 Pro        TRUE     
6 Beginner   TRUE     
  hashed_email                                                     played_hours
1 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d 56.1        
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686 53.9        
3 a175d4741dc84e6baf77901f6e8e0a06f54809a34e6b5211159bced346f7fb3e 48.4        
4 24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe 32.0        
5 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
6 9782452b8e41d1f140374eec34c84f464df6ed12e2777e5b4f7150912b8ab78c 23.7        
  name   gender age
1 Dana   Male   23 
2 Alex   Male   17 
3 Xander Female 17 
4 Sidney Male   22 
5 Morgan Male    9 
6 Isla   Male   24